### PR TITLE
Use stream

### PR DESCRIPTION
**Pull request by marcmognol**
**Changes**
Use stream instead of local file

**Motivation**

Fix Issue #21

### CODE REVIEW

**Comment by jfversluis** [comment]
Thanks so much for this @marcmognol ! We'll have a look at it ASAP!

**Comment by bijington** [comment]
Yes thank you @marcmognol. I have one initial question - do we need the interim `MemoryStream`? Could we just stream from whatever the consumer provides? I appreciate not all streams support seeking, etc. so this is very much asking for opinions

**Comment by marcmognol** [comment]
Hi @jfversluis, @bijington,

I've tried to use Stream instead of MemoryStream but I have a NotSupportedException while playing audio.
If you have any idea !

Marc

File: src/Plugin.Maui.Audio/AudioPlayer.android.cs [hunk 1]
```diff
@@ -40,44 +38,18 @@ public bool Loop
...
-                //return false;
-            }
-        }
 
-        player.Prepare();
-    }
+		stream = new MemoryStream();
+		audioStream.CopyTo(stream);
+		var mediaDataSource = new StreamMediaDataSource(stream);
+		player.SetDataSource(mediaDataSource);
```
Comment (marcmognol) [inline src/Plugin.Maui.Audio/AudioPlayer.android.cs:50]: This call is only supported on "android 23.0 and later". What do you think ?

Comment (marcmognol) [reply src/Plugin.Maui.Audio/AudioPlayer.android.cs]: Sure, I've restore previous code and add condition.

File: src/Plugin.Maui.Audio/AudioPlayer.android.cs [hunk 2]
```diff
@@ -8,10 +8,11 @@ namespace Plugin.Maui.Audio;
 partial class AudioPlayer : IAudioPlayer
 {
     readonly MediaPlayer player;
-    static int index = 0;
+    int index = 0;
     double volume = 0.5;
     double balance = 0;
     string path = string.Empty;
+    readonly MemoryStream stream;
```
Comment (jfversluis) [inline src/Plugin.Maui.Audio/AudioPlayer.android.cs:15]:
```suggestion
    readonly MemoryStream? stream;
```

**Review by jfversluis — APPROVED** [review]
Looks good to me! @bijington any opinions?

**Comment by bijington** [comment]
> Looks good to me! @bijington any opinions?

Sorry I've been rubbish at getting things done recently! I'll make a note to look this week

**Review by bijington — APPROVED** [review]

**Comment by bijington** [comment]
1 week turned into 27 days somehow but it looks good to me! Thanks @marcmognol

**Comment by bijington** [comment]
> Hi @jfversluis, @bijington,
> 
> I've tried to use Stream instead of MemoryStream but I have a NotSupportedException while playing audio. If you have any idea !
> 
> Marc

Sorry I never managed to investigate this, I wonder if it relates to being able to Seek or not. Happy to investigate this as a further enhancement outside of this PR though

**Pull request merged**